spec/controllers/tag_set_nominations_controller_spec.rb

```diff
@@ -0,0 +1,1547 @@
+                    get :index, tag_set_id: owned_tag_set.id
+                    expect(assigns(:nominations)[:character]).to be_nil
+                  end
+                end
+
+                context 'relationship_nomination_limit is 0' do
+                  it 'does not return relationship nominations' do
+                    add_unreviewed_relationship_nominations(1)
+                    owned_tag_set.update_column(:relationship_nomination_limit, 0)
+                    get :index, tag_set_id: owned_tag_set.id
+                    expect(assigns(:nominations)[:relationship]).to be_nil
+                  end
+                end
+
+                def add_unreviewed_character_nominations(num)
+                  num.times do |i|
+                    CharacterNomination.create(tag_set_nomination: tag_set_nomination, fandom_nomination: fandom_nom,
+                                               tagname: "New Character #{i}")
+                  end
+                end
+
+                def add_unreviewed_relationship_nominations(num)
+                  num.times do |i|
+                    RelationshipNomination.create(tag_set_nomination: tag_set_nomination, fandom_nomination: fandom_nom,
+                                                  tagname: "New Relationship #{i}")
+                  end
+                end
+              end
+            end
+          end
+
+          context 'logged in user is not moderator' do
+            let(:user) { random_user }
+
+            it 'redirects and returns an error message' do
+              get :index, tag_set_id: owned_tag_set.id
+              it_redirects_to_with_error(tag_sets_path, "You can't see those nominations, sorry.")
+            end
+          end
+        end
+
+        # TODO: how can OwnedTagSet.find not raise an error but still return falsey?
+        xcontext 'no tag set' do
+          let(:user) { random_user }
+
+          it 'redirects and returns an error message' do
+            get :index, tag_set_id: nil
+            it_redirects_to_with_error(tag_sets_path, 'What nominations did you want to work with?')
+          end
+        end
+      end
+    end
+  end
+
+  describe 'GET show' do
+    context 'user is not logged in' do
+      it 'redirects and returns an error message' do
+        get :show, id: tag_set_nomination.id, tag_set_id: owned_tag_set.id
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you " +
+            "were trying to reach. Please log in.")
+      end
+    end
+
+    context 'user is logged in' do
+      context 'invalid params' do
+        before do
+          fake_login_known_user(tag_nominator)
+        end
+
+        # TODO: how can OwnedTagSet.find not raise an error but still return falsey?
+        xcontext 'no tag set' do
+          it 'redirects and returns an error message' do
+            get :show, id: tag_set_nomination.id, tag_set_id: nil
```

Comment:
This one already has the right error message to test load_tag_set, no?

For this case (and most of the other cases I marked with TODOs), I marked the tests as pending because I'm not sure how it can get to this code path. The code is set up basically with:
```ruby
if OwnedTagSet.find(blah)
  do_a_thing
else
  flash[:error] = "Oh no."
end
```

But my understanding is that if OwnedTagSet.find fails, it would raise an ActiveRecord:RecordNotFound error, and never make it into the else code path. Is that wrong? Do you have suggestions for testing the else case in this situation?